src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -873,26 +920,20 @@
   [{:keys [projections]}]
   (->> projections
        (remove (comp :query-only? val))
```

Comment:
It's actually not a kv pair in the projected-fields case.  The projected fields are a collection, not a map of two element vectors (i.e. projection-schemas), whose order in an in clause parallels the columns'.